irma-core/index.js

```diff
@@ -44,66 +42,27 @@ module.exports = class IrmaCore {
     this._modules.filter(m => m.stateChange)
                  .forEach(m => m.stateChange(state));
 
-    const {newState, payload, isFinal} = state;
+    const {newState, isFinal} = state;
 
-    switch(newState) {
-      case 'Success':
-        this._close().then(result => {
-          if ( this._resolve ) this._resolve(result);
-        });
-        break;
-      case 'MediumContemplation':
-        if (this._userAgentIsMobile())
-          this._stateMachine.transition('showIrmaButton', this._getSessionUrls(payload));
-        else
-          this._stateMachine.transition('showQRCode', this._getSessionUrls(payload));
-        break;
-      default:
-        if ( isFinal ) {
-          this._close().then(result => {
-            if ( this._reject ) result ? this._reject(result) : this._reject(newState);
-          });
-        }
-        break;
+    if (isFinal) {
+      let func = newState == 'Success' ? this._resolve : this._reject;
+      this._close().then(result => {
+        if ( func ) result ? func(result) : func(newState);
+      });
     }
   }
 
   _close() {
-    return this._modules.filter(m => m.close)
-      .reduce(
-        (prev, m) => prev.then(returnValues => m.close().then(res => {
-          if (res) returnValues.push(res)
-          return returnValues;
-        })),
-        Promise.resolve([])
-      )
-      .then(returnValues => returnValues.length > 1 ? returnValues : returnValues[0]);
-  }
-
-  _userAgentIsMobile() {
-    return this._userAgent =='Android' || this._userAgent == 'iOS';
-  }
-
-  _getSessionUrls(sessionPtr) {
-    let json = JSON.stringify(sessionPtr);
-    let universalLink = `https://irma.app/-/session#${encodeURIComponent(json)}`;
-    let mobileLink;
-    switch (this._userAgent) {
-      case 'Android':
-        // Universal links are not stable in Android webviews and custom tabs, so always use intent links.
-        let intent = `Intent;package=org.irmacard.cardemu;scheme=irma;l.timestamp=${Date.now()}`;
-        let fallback = `S.browser_fallback_url=${encodeURIComponent(universalLink)}`;
-        mobileLink =  `intent://qr/json/${encodeURIComponent(json)}#${intent};${fallback};end`;
```

Comment:
The different URL options should I think move back to here, since this is really core IRMA.